### ISSUE TITLE
Access token not automatically refreshed

### ISSUE THREAD

**Issue by kpine** [python-wink/python-wink#102]
Noticed my wink components weren't being updated in Home Assistant. I saw some wink.py exceptions in the logs, but ignored them. But after a restart saw this error:
```
dietpi@hasspi:/home/homeassistant/.homeassistant$ cat home-assistant.log
2018-02-15 23:58:18 ERROR (MainThread) [homeassistant.setup] Error during setup of component wink
Traceback (most recent call last):
  File "/home/homeassistant/.pyenv/versions/3.6.3/envs/homeassistant-3.6.3/lib/python3.6/site-packages/homeassistant/setup.py", line 199, in_async_setup_component
    component.setup, hass, processed_config)
  File "/home/homeassistant/.pyenv/versions/3.6.3/lib/python3.6/asyncio/futures.py", line 332, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/home/homeassistant/.pyenv/versions/3.6.3/lib/python3.6/asyncio/tasks.py", line 250, in _wakeup
    future.result()
  File "/home/homeassistant/.pyenv/versions/3.6.3/lib/python3.6/asyncio/futures.py", line 245, in result
    raise self._exception
  File "/home/homeassistant/.pyenv/versions/3.6.3/lib/python3.6/concurrent/futures/thread.py", line 56, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/homeassistant/.pyenv/versions/3.6.3/envs/homeassistant-3.6.3/lib/python3.6/site-packages/homeassistant/components/wink/__init__.py", line 312, in setup
    pywink.get_subscription_key())
  File "/home/homeassistant/.pyenv/versions/3.6.3/envs/homeassistant-3.6.3/lib/python3.6/site-packages/pywink/api.py", line 577, in get_subscription_key
    response_dict = wink_api_fetch()
  File "/home/homeassistant/.pyenv/versions/3.6.3/envs/homeassistant-3.6.3/lib/python3.6/site-packages/pywink/api.py", line 598, in wink_api_fetch
    raise WinkAPIException("401 Response from Wink API.  Maybe Bearer token is expired?")
pywink.api.WinkAPIException: 401 Response from Wink API.  Maybe Bearer token is expired?
```
Looked around in the python-wink source code and saw some refresh code had been added in the past. I was able to refresh the token via the API and update my .wink.conf file.
```
>>> pywink.set_wink_credentials(client_id, client_secret, access_token=access_token, refresh_token=refresh_token)
>>> lights = pywink.get_light_bulbs()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/homeassistant/.pyenv/versions/homeassistant-3.6.3/lib/python3.6/site-packages/pywink/api.py", line 445, in get_light_bulbs
    return get_devices(device_types.LIGHT_BULB)
  File "/home/homeassistant/.pyenv/versions/homeassistant-3.6.3/lib/python3.6/site-packages/pywink/api.py", line 610, in get_devices
    ALL_DEVICES = wink_api_fetch(end_point)
  File "/home/homeassistant/.pyenv/versions/homeassistant-3.6.3/lib/python3.6/site-packages/pywink/api.py", line 598, in wink_api_fetch
    raise WinkAPIException("401 Response from Wink API.  Maybe Bearer token is expired?")
pywink.api.WinkAPIException: 401 Response from Wink API.  Maybe Bearer token is expired?
>>> pywink.refresh_access_token()
'redacted'
>>> lights = pywink.get_light_bulbs()
>>>
```
Maybe this is an issue/feature that should be solved in Home Assistant instead, but I thought I'd try here first.

**Comment by w1ll1am23** [comment]
Thanks I'll take a look it is an issue here for sure. It should be refreshing automatically in the library.

**Comment by w1ll1am23** [comment]
So I am going to add in some code to call the refresh_access_token() from the wink_api_fetch() method but I am confused why that is needed. Currently if the API call fails refresh_access_token() is automatically called, then Home Assistant should periodically and on shutdown, pull the current access token and update it in the .wink.conf it sounds like that wasn't the case for you? You said you saw some Wink errors in the log and then rebooted, do you remember what the errors were?

**Comment by kpine** [comment]
Sorry for the late reply, I was offline for a while. I don't have any additional logs except for what's above. The HA logs I did have rotated out since then and I don't have any recent wink errors. Another thing I noticed, the timestamp on my .wink.conf is being updated, and the wink components are now being shown in HA, so I assume it's doing the right thing. If you want, please go ahead and close this and I can re-open if I run into it again and have any useful information.

**Comment by kpine** [comment]
Another thing I did notice. There may be an issue with home assistant, not here. My wink climate component is logging to an InfluxDb server I have offsite, using the HA influx component. About the time I left for vacation until today, I have no events stored in InfluxDB. That was also the time I noticed the Wink component was not working. Since I restarted HA, the events are being logged in InfluxDb. However, the HA History page is showing this data in its own graphs.
